include/ultra64/controller.h

```diff
@@ -119,6 +119,12 @@
 #define BTN_B           0x4000
 #define BTN_A           0x8000
 
+#ifdef __GNUC__
+// Ensure data cache coherency for OSPifRam structures by aligning to the data cache line size.
+// On older compilers such as IDO this was done by placing each OSPifRam at the top of the file it is declared in,
+// however file alignment should not be relied on in general.
+__attribute__((aligned(0x10)))
```

Comment:
what happens if `aligned(16)` is set on this struct with IDO?

Reply:
I'm not sure what you mean specifically, `__attribute__(x)` isn't understood by IDO

Reply:
`__attribute__` is a GNU extension ( https://gcc.gnu.org/onlinedocs/gcc/Attribute-Syntax.html )
IDO doesn't know about it, it would be a syntax error

This isn't needed with IDO due to a combination of
- segments being multiples-of-0x10-aligned (cf spec / mkldscript),
- sections sizes being 0x10-aligned by IDO (with zero padding),
- placing the relevant data at the beginning of files,
- and IDO not reordering data (so the relevant data is at the beginning of the object file section) (actually idk if this holds true for bss which this may be placed in hmm...)

Reply:
I didn't used the macro here since it's a libultra header while `alignment.h` isn't, so it would be incorrect to include it in this file